tests/grids/test_grid_refinement.py

```diff
@@ -17,104 +15,104 @@
-
-            super().__init__(pts, tri, "GridWithTwoCells")
-            self.history = ["TwoGrid"]
+class TestGridRefinement2DSimplex:
+    @pytest.fixture
+    def one_cell_grid(self):
+        g = TriangleGrid(
+            np.array([[0, 1, 0], [0, 0, 1], [0, 0, 0]]),
+            np.array([[0], [1], [2]]),
+            "GridWithOneCell",
+        )
+        setattr(g, "history", ["OneGrid"])
+        return g
+
+    @pytest.fixture
+    def two_cell_grid(self):
+        g = TriangleGrid(
+            np.array([[0, 1, 0, 1], [0, 0, 1, 1], [0, 0, 0, 0]]),
+            np.array([[0, 1], [1, 3], [2, 2]]),
+            "GridWithTwoCells",
+        )
+        setattr(g, "history", ["TwoGrid"])
+        return g
 
-    def test_refinement_single_cell(self):
-        g = self.OneCellGrid()
+    def test_refinement_single_cell(self, one_cell_grid):
+        g = one_cell_grid
         h, parent = refinement.refine_triangle_grid(g)
         h.compute_geometry()
 
-        self.assertTrue(h.num_cells == 4)
-        self.assertTrue(h.num_faces == 9)
-        self.assertTrue(h.num_nodes == 6)
-        self.assertTrue(h.cell_volumes.sum() == 0.5)
-        self.assertTrue(np.allclose(h.cell_volumes, 1 / 8))
+        assert h.num_cells == 4
+        assert h.num_faces == 9
+        assert h.num_nodes == 6
+        assert h.cell_volumes.sum() == 0.5
+        assert np.allclose(h.cell_volumes, 1 / 8)
 
         known_nodes = np.array([[0, 0.5, 1, 0.5, 0, 0], [0, 0, 0, 0.5, 1, 0.5]])
         test_utils.compare_arrays(h.nodes[:2], known_nodes)
-        self.assertTrue(np.all(parent == 0))
+        assert np.all(parent == 0)
 
-    def test_refinement_two_cells(self):
-        g = self.TwoCellsGrid()
+    def test_refinement_two_cells(self, two_cell_grid):
+        g = two_cell_grid
         h, parent = refinement.refine_triangle_grid(g)
         h.compute_geometry()
 
-        self.assertTrue(h.num_cells == 8)
-        self.assertTrue(h.num_faces == 16)
-        self.assertTrue(h.num_nodes == 9)
-        self.assertTrue(h.cell_volumes.sum() == 1)
-        self.assertTrue(np.allclose(h.cell_volumes, 1 / 8))
+        assert h.num_cells == 8
+        assert h.num_faces == 16
+        assert h.num_nodes == 9
+        assert h.cell_volumes.sum() == 1
+        assert np.allclose(h.cell_volumes, 1 / 8)
 
         known_nodes = np.array(
             [[0, 0.5, 1, 0.5, 0, 0, 1, 1, 0.5], [0, 0, 0, 0.5, 1, 0.5, 0.5, 1, 1]]
         )
         test_utils.compare_arrays(h.nodes[:2], known_nodes)
-        self.assertTrue(np.sum(parent == 0) == 4)
-        self.assertTrue(np.sum(parent == 1) == 4)
-        self.assertTrue(np.allclose(np.bincount(parent, h.cell_volumes), 0.5))
+        assert np.sum(parent == 0) == 4
+        assert np.sum(parent == 1) == 4
+        assert np.allclose(np.bincount(parent, h.cell_volumes), 0.5)
 
 
-class TestRefinementMortarGrid(unittest.TestCase):
+class TestRefinementMortarGrid:
     def test_mortar_grid_1d(self):
```

Comment:
Despite the name of this class, I think these tests should rather go in test_mortar_grid, since they mainly check that the mortar mappings are correctly updated. Does this make sense?

Reply:
So this method `test_mortar_grid_1d` should not be in this class, but a test as an independent function?

Reply:
I put this test into `tests/grids/test_mortar_grid.py`